### ISSUE TITLE
W: Lint/UselessAssignment: - False Positive on recursive method with incrementer (+=1)

### ISSUE THREAD

**Issue by luke-hill** [rubocop-hq/rubocop#6846]
Apologies if this isn't a bug. But it feels like one. I've tried removing the variable and leaving `+=1` which doesn't work, and I've tried underscoring it, which doesn't work as I'd want to because the new variable doesn't exist in the scope.

## Expected behavior
Not complain about a variable that has usage

## Actual behavior
Running rubocop on the `scratch.rb` file below throws a warning for useless variables.

```
W: Lint/UselessAssignment: Useless assignment to variable - tries. (https://github.com/rubocop-hq/ruby-style-guide#underscore-unused-vars)
      looper(tries += 1)
          ^^^^^
```

## Steps to reproduce the problem

```
# scratch.rb

def looper(tries = 0)
  return if tries >= 3
  puts tries
  sleep tries
  looper(tries += 1)
end

looper
```
## RuboCop version

```
0.65.0 (using Parser 2.6.0.0, running on ruby 2.6.1 x86_64-linux)
```

**Comment by twalpole** [comment]
It’s not complaining about a useless variable, it’s complaining about a useless assignment.. The assignment is useless because `tries` is local to the method, and the method returns without using the variable again after the assignment (the recursive call to looper creates a new instance of the variable) The fix for the warning is 

    looper(tries + 1)

**Comment by luke-hill** [comment]
![](https://media.giphy.com/media/4pMX5rJ4PYAEM/giphy.gif)
